packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -1,34 +1,69 @@
 import 'dart:convert';
 
 import 'package:meta/meta.dart';
-
+import './authorization_request.dart';
+
+/// Authorization details from a successful Sign in with Apple flow.
+///
+/// Most fields are optional in this class.
+///
+/// Especially [givenName], [familyName], and [email] member will only be provided on the first authorization between
+/// the app and Apple ID.
+///
+/// The [authorizationCode] member is always present and should be used to check the authorizations with Apple servers
+/// from your backend. Upon successful validation, you should create a session in your system for the current user,
+/// or consider her now logged in.
 @immutable
-abstract class AuthorizationCredential {}
-
-class AuthorizationCredentialAppleID implements AuthorizationCredential {
-  AuthorizationCredentialAppleID({
+class AuthorizationCredentialAppleID {
+  /// Creates an instance which contains the result of a successful Sign in with Apple flow.
+  const AuthorizationCredentialAppleID({
     @required this.userIdentifier,
     @required this.givenName,
     @required this.familyName,
     @required this.email,
-    @required this.identityToken,
     @required this.authorizationCode,
   }) : assert(authorizationCode != null);
 
+  /// An identifier associated with the authenticated user.
+  ///
+  /// This will be provided upon every sign-in.
```

Comment:
I guess this statement is contradictory to the `Can be null` in the end.

```suggestion
  /// This will always be provided on iOS and macOS systems. On Android, however, this will not be present.
```